genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/SetUpJobAction.java

```diff
@@ -504,4 +482,49 @@ private File createJobEnvironmentFile(
         return Collections.unmodifiableMap(env);
     }
 
+    private void cleanupJobDirectory(
+        final File jobDirectory,
```

Comment:
General comment: 

Perhaps instead of using the old `java.io.File` classes we should try defaulting to using the `java.nio.Path` as the default and save us from creating paths all over?

Reply:
Migrated to `Path` since it was indeed converting to it everywhere..

Reply:
Also, didn't realize `File` is the old way and `Path` is the new one, will keep this in mind going forward.